### DISCUSSION TITLE
Remove the persistent notifications for unreachable devices

### DISCUSSION THREAD

**Discussion by eikowagenknecht** [SukramJ/aiohomematic#161] [Ideas]
Taking the discussion from https://github.com/danielperna84/hahomematic/issues/157 to here as @SukramJ suggests that this is a feature request and not a bug:

As of now, unreachable devices generate a persistent notification:
![image](https://user-images.githubusercontent.com/1475672/148908787-1ef437fe-ea57-48c7-b410-bd66b5edec72.png)

This leads to many notifications for flakey devices which is undesirable. Users can create automations to generate this notifications, but they can not automatically remove it.

The usual way would be to instead just mark the entities as "not reachable" as it is done in the old homematic integration and all other HA integrations I know.

I would strongly suggest to handle it this way as well in the new integration.

**Comment by danielperna84** [comment]
As mentioned in the issue, some users previously requested to have more detail on service messages of the CCU. With the old integration we only have the number of service messages, but nothing about their content. The way it is now allows users to only look in the HA UI to know what the problem is. Depending on how your HA UI looks you might not notice straight away which devices are causing problems. And it's more work to generate the notifications automatically. So based on my personal taste, I vote to keep it this way.

**Reply by emufan** [reply]
I would vote to see this persistent notification disabled/removed as well. The user, who needs it, can create them via automation.

My setup: I have an outdoor sensor and know, that this has sometime problems. But that doesn't matter, some seconds afterwards it is back again, so I don't care. But the persistent notification is generated.

If such notifications appear, I'm used to thinking, that there is really something important and look (during usage of UI, app, ...), but currently most of the time it is only the unimportant sporadic unreachable of the very unimportant light sensor.

**Reply by eikowagenknecht** [reply]
I see why some users might want that. But for others (like @emufan and me) it adds a false sense of urgency for sporadic issues with sensors we already know about. And like always: When you see too many warnings / errors, the really important ones get lost in the noise.

Maybe there is another way to achive what the users @danielperna84 talks about want:
- Log a warning in HA log for unreachable devices (many integrations do this)
- Create a blueprint that responds to this and create persistant notifications from it

The result would look the same as now, but optional. And writing it as a warning in the HA log seems like a good idea anyways if you say it's important enough to even create a notification out of it ;-)

This way it would be relatively easy for users to get this functionality (import the bluepint and say which devices should be "monitored") while not forcing it on everyone. The blueprint could be linked in the Integration documentation with one of those fancy "My ..." links.

WDYT?

**Reply by noxhirsch** [reply]
It's "good" to hear that I'm not the only one with this problem. I thought it was due to my apartment and the probably quite busy frequencies in downtown Mannheim. Especially after a reboot of HA, I get a few notifications (~1-3) which looks like a config error until I open the notification sidebar.
As a workaround I had created an automation which kills these notifications instantly, but a solution via a log entry and triggering an event + blueprint I'd also find much better.

**Reply by danielperna84** [reply]
I don't know if the method using a blueprint would work. Service messages are polled, like the variables. There are no events we can use like we do for the entities. This all happens in the code, and I don't think blueprints can do anything here.

If we would just generate an event (which could be used for automations) as soon as we see service messages, this would fire the event again and again for every poll if we don't keep track of the known messages. But if we keep track, that could be an option to handle this issue.

@SukramJ 

What do you think about us generating our own events for service messages so users can handle them however they want?

**Reply by SukramJ** [reply]
ServiceMessages are a mess with CCU and not really useable. Here we use the events `UNREACH` und `STICKS_UNREACH` to display the persistent notifications. The usable event for blueprints is the state switch of each entitties `available` flag.

**Reply by SukramJ** [reply]
Maybe someone who is willing to ged rid of persistent notification is willing to create a useable blueprint.

**Reply by eikowagenknecht** [reply]
Sure, I can have a look at creating a blueprint that converts the information in the available flag to a persistant notification if that's all that's needed.

**Reply by SukramJ** [reply]
A blueprint is no longer needed. [See](https://github.com/danielperna84/custom_homematic/blob/devel/blueprints/automation/homematicip_local_persistent_notification.yaml)
(Edit:just saw that half of my answer is missing)
The Blueprint is a 1 to 1 replacement for the automatic notification. To make the BP easier I also adjusted the CC. 

@eikowagenknecht I hope you haven't had too much effort yet.
